src/openfermion/transforms/_conversion_test.py

```diff
@@ -507,3 +513,374 @@ def test_get_fermion_operator_majorana_operator():
 def test_get_fermion_operator_wrong_type():
     with pytest.raises(TypeError):
         _ = get_fermion_operator(QubitOperator())
+
+
+class GetNumberPreservingSparseOperatorIntegrationTestLiH(unittest.TestCase):
+
+    def setUp(self):
+        # Set up molecule.
+        geometry = [('Li', (0., 0., 0.)), ('H', (0., 0., 1.45))]
+        basis = 'sto-3g'
+        multiplicity = 1
+        filename = os.path.join(THIS_DIRECTORY, 'data',
+                                'H1-Li1_sto-3g_singlet_1.45')
+        self.molecule = MolecularData(
+            geometry, basis, multiplicity, filename=filename)
+        self.molecule.load()
+
+        # Get molecular Hamiltonian.
+        self.molecular_hamiltonian = self.molecule.get_molecular_hamiltonian()
+
+
+        def get_reference_state(
+                num_qubits,
+                num_electrons,
+                spin_preserving=True,
+                reference_determinant=None,
+                excitation_level=None,
+                do_print=False):
+
+            if (reference_determinant is None):
+                unoccupied = [False for i in range(num_qubits - num_electrons)]
+                occupied = [True for i in range(num_electrons)]
+
+                reference_determinant = numpy.asarray(occupied + unoccupied)
+            else:
+                reference_determinant = numpy.asarray(reference_determinant)
+
+            if (excitation_level is None):
+                excitation_level = num_electrons
+            else:
+                excitation_level = min(excitation_level, num_electrons)
+
+
+            state_array = numpy.asarray(list(_iterate_basis_(
+                reference_determinant, excitation_level, spin_preserving)))
+
+            space_size = state_array.shape[0]
+            if do_print:
+                print("Dimension of Hilbert space for sparse operator = %d" % space_size)
+
+            reference = numpy.zeros((space_size))
+            reference[0] = 1.0
+
+            return reference
+
+        self.get_reference_state = get_reference_state
```

Comment:
Why not just define this function as a method of the class?

Reply:
You know, this isn't really doing anything anymore except just making a vector [1.0, 0.0, ... , 0.0] so I just put that code in the tests where I used it and removed this entirely.